### ISSUE TITLE
[BUG] - New v4 Diametral Solid Cylinder Field has a singularity at the axis :(

### ISSUE THREAD

**Issue by OrtnerMichael** [magpylib/magpylib#370]
```
import magpylib as magpy
src = magpy.magnet.Cylinder((1,0,0), (2,2))
B = src.getB((0,0,5))
```

must be caught as a special case - should be an easy fix

**Comment by OrtnerMichael** [comment]
Unfortunately the indeterminate form at r=0 is numerically highly instable.

![image](https://user-images.githubusercontent.com/47384012/127970573-2fcfbcb0-b183-44dc-8f08-74452262e8e0.png)

**Comment by OrtnerMichael** [comment]
this is a long-term goal which is now addressed in #374
